app/graph.py

```diff
@@ -441,7 +457,9 @@ def update_node(self, apic, node):
     def update(self):
```

Comment:
It might be worth refactoring this function into parts because now its quite large and complex.

Reply:
Will revisit this in the next iteration for optimization and refactoring